climada/util/plot.py

```diff
@@ -290,6 +292,98 @@ def geo_im_from_array(array_sub, coord, var_name, title,
 
     return axes
 
+def geo_scatter_categorical(array_sub, geo_coord, var_name, title,
+                            cat_name=None, pop_name=True, buffer=BUFFER,
+                            extend='neither', proj=ccrs.PlateCarree(),
+                            shapes=True, axes=None, **kwargs):
+    """Map plots for categorical data defined in array(s) over input
+    coordinates. The categories must be a finite set of unique
+    values as can be identified by np.unique() (int, float, strings, ...).
+    
+
+    Parameters:
+        array_sub (np.array(1d or 2d) or list(np.array)): Each array (in a row
+            or in  the list) are categorical values at each point
+            in corresponding geo_coord that are ploted in one subplot.
+        coord (2d np.array): (lat, lon) for each point in a row.
+            The same grid is used for all subplots.
+        var_name (str or list(str)): label to be shown in the colorbar. If one
+            provided, the same is used for all subplots. Otherwise provide as
+            many as subplots in array_sub.
+        cat_name (list(str)), optional: List of category names in same order
+            as the ordered categories obtain from np.unique(array_sub). All
+            subplots are assumed to have the same categories. 
+            Default is numbers (1,2,...)
+        title (str or list(str)): subplot title. If one provided, the same is
+            used for all subplots. Otherwise provide as many as subplots in
+            array_sub.
+        proj (ccrs): coordinate reference system used in coordinates
+        smooth (bool, optional): smooth plot to RESOLUTIONxRESOLUTION. Default:
+            True.
+        kwargs (optional): arguments for pcolormesh matplotlib function.
+
+    Returns:
+        cartopy.mpl.geoaxes.GeoAxesSubplot
+    """
+    
+    # Generate array of values used in each subplot
+    num_im, list_arr = _get_collection_arrays(array_sub)
+    list_tit = to_list(num_im, title, 'title')
+    list_name = to_list(num_im, var_name, 'var_name')
+    list_coord = to_list(num_im, geo_coord, 'geo_coord')
+    
+    #default cmap
+    cmap_name = 'Accent'
+    if 'cmap' in kwargs:
+        cmap_name = kwargs['cmap']
+        del kwargs['cmap']
```

Comment:
This change is needed because the `cmap` is now specified explicitly in line 369 so that the new discrete cmap can be used with its normalization factor. 

```
hex_bin = axis.scatter(coord[:, 1], coord[:, 0], c=valcat, norm=norm,
                               cmap=cmap, transform=proj, **kwargs)
```